New Radix Themes release and documentation tweaks

vladmoroz opened the pull request:
- Updates Radix Themes and Radix Colors to the upcoming versions
- Some documentation tweaks across the board, mostly around how we showcase some component variants and high contrast options
- Some design tweaks on the demos
- Scroll active sidebar link into view
- Clean up Radix Themes example UIs

andy-hook reviewed (commented):
I diff'd against both sites for all comps, the home page, and the colors pages. Everything looks great overall with some really nice improvements in the demos.

Left couple of minor comments and I noticed this "Invoice paid" title feels a bit tight now:

![CleanShot 2023-09-29 at 15 45 03](https://github.com/radix-ui/website/assets/11708259/f2bd6cec-3c9d-41da-bb35-7a81640fd165)

I haven't checked the accuracy of the release notes as frankly I haven't kept up there.

vladmoroz commented on data/themes/docs/theme/color.mdx:
```diff
@@ -35,50 +35,58 @@ Alpha variants are used automatically within components so no configuration is r
+
+### Aliasing colors
+
+You may find that certain shades that have less common names pair better with your main accent color. For example, it is common to use `crimson` and `ruby` in place of `red`, or `teal` and `jade` in place of `green`.
+
+To reclaim the more intuitive color names like `red`, `green`, or `blue`, among others, you can remap Radix Themes tokens in place of one another:
```
Yes! First part is better and to the point. I liked the "reclaim" bit because I felt that it pinpointed what you are doing exactly, but not feeling strongly about that one as I have the curse of knowledge here. What do you think about this?

```suggestion
You may prefer to use generic color names to refer to the color shades that you want to use. For example, it is common to refer to `crimson`, `jade` and `indigo` as `red`, `green` and `blue`. 

In this case, you can remap Radix Themes tokens in place of one another to reclaim the more intuitive color names like `red`, `green`, or `blue`, among others.
```

> I personally found the "less common names" part kind of confusing and I wasn't sure what was being explained at first.

I was trying to get at how sometimes "red" might be an objectively worse choice than "crimson" / "ruby" etc depending on your main accent, but we can lose that part.

andy-hook replied:
Works for me, perhaps we could avoid the color name repetition on the last line though?

> In this case, you can remap Radix Themes tokens in place of one another to reclaim such naming:

(or something)

vladmoroz replied:
"In this case, you can remap Radix Themes tokens in place of one another and reclaim the color names you want to use"?

vladmoroz commented on data/themes/docs/theme/color.mdx:
```diff
@@ -35,50 +35,58 @@ Alpha variants are used automatically within components so no configuration is r
+
+In this example, using the `red` color in Radix Themes components and tokens would now reference the original `ruby` scale.
+
+This is also how Radix Themes re-maps tinted grays like `mauve`, `slate`, `sage`, etc. to `gray` according to your theme setting, so that `gray` always refers to the shade you picked.
```
OK, will remove

vladmoroz commented:
> I diff'd against both sites for all comps, the home page, and the colors pages. 

Bless you, appreciate that a ton.

andy-hook reviewed (approved):
Great job, approving with the caveat I mentioned on the change log stuff.

vladmoroz commented:
Released 2.0.0 so going to merge now.

I added in some juicy stuff on top of what you've reviewed @andy-hook, feel free to catch up as you guys get back. Main additions:
- Beef up Radix Colors homepage
- Add links to previous versions of Radix Themes docs
- Improve how favicon responds to theme changes
- Do a pass on meta titles
- Set homepage meta to make sense for all of Radix UI. As of now it doesn't:
  <img width="907" alt="image" src="https://github.com/radix-ui/website/assets/8441036/fd7fecff-d7e0-4b70-99fc-ce83c8ed1d88">
- Docs tweaks
  - Added container widths
  - Tweaked breakpoint descriptions
  - Remove `/colors/docs/tests` (almost irrelevant now with the new homepage & a lot of the content there was questionable)
  - Replace old alert components with callouts
